src/redux/slices/userSlice.js

```diff
@@ -1,20 +1,288 @@
+      rejectWithValue(err);
+    }
+  }
+);
+
+export const removeBan = createAsyncThunk(
+  'user/removeBan',
+  async (userId, { rejectWithValue }) => {
+    try {
+      const { token, user } = await checkToken();
+
+      const res = await axios.put(
+        API_URL + `/api/user/${userId}`,
+        { status: 'inactive' },
+        { headers: { authorization: token } }
+      );
+
+      const updatedUser = res.data;
+
+      if (!updatedUser.id) throw new Error('Failed to update user status');
+
+      return updatedUser;
+    } catch (err) {
+      rejectWithValue(err);
+    }
+  }
+);
+
 const userSlice = createSlice({
-  name: 'users',
+  name: 'user',
   initialState: {
-    users: [],
     user: {},
+    error: '',
+    isLoading: false,
   },
-  reducers: {
-    testUser: (state) => {
-      state.user = { test: 'user' };
-    },
+  reducers: {},
+  extraReducers: (builder) => {
+    builder
+      .addCase(fetchUser.fulfilled, (state, { payload }) => {
+        state.user = payload;
+        state.isLoading = false;
+        state.error = '';
+      })
+      .addCase(fetchUser.pending, (state, { payload }) => {
+        state.user = {};
+        state.isLoading = true;
+        state.error = '';
+      })
+      .addCase(fetchUser.rejected, (state, action) => {
+        state.user = {};
+        state.isLoading = false;
+        state.error = action.payload.message;
+      })
+      .addCase(createNewUser.fulfilled, (state, { payload }) => {
+        state.user = payload;
+        state.isLoading = false;
+        state.error = '';
+      })
+      .addCase(createNewUser.pending, (state, { payload }) => {
+        state.user = {};
+        state.isLoading = true;
+        state.error = '';
+      })
+      .addCase(createNewUser.rejected, (state, action) => {
+        state.user = {};
+        state.isLoading = false;
+        state.error = action.payload.message;
+      })
+      .addCase(updateUser.fulfilled, (state, { payload }) => {
+        state.user = payload;
+        state.isLoading = false;
+        state.error = '';
+      })
+      .addCase(updateUser.pending, (state, { payload }) => {
+        state.user = {};
+        state.isLoading = true;
+        state.error = '';
+      })
+      .addCase(updateUser.rejected, (state, action) => {
+        state.user = {};
+        state.isLoading = false;
+        state.error = action.payload.message;
+      })
+      .addCase(updateLocation.fulfilled, (state, { payload }) => {
+        state.user = payload;
+        state.isLoading = false;
+        state.error = '';
+      })
+      .addCase(updateLocation.pending, (state, { payload }) => {
+        state.user = {};
+        state.isLoading = true;
+        state.error = '';
+      })
+      .addCase(updateLocation.rejected, (state, action) => {
+        state.user = {};
+        state.isLoading = false;
+        state.error = action.payload.message;
+      })
+      .addCase(deleteUser.fulfilled, (state, { payload }) => {
```

Comment:
Can a user delete their own account or can this only be done by an admin. If it's an admin thing, should the user be emptied in the store?

Reply:
per our conversation -- removed delete thunk entirely